src/java.base/share/classes/java/lang/reflect/Field.java

```diff
@@ -906,8 +971,14 @@ public void setByte(Object obj, byte b)
             Class<?> caller = Reflection.getCallerClass();
             checkAccess(caller, obj);
             getFieldAccessor().setByte(obj, b);
+            return;
+        }
+
+        FieldAccessor fa = getOverrideFieldAccessor();
+        if (!Modifier.isFinal(modifiers)) {
+            fa.setByte(obj, b);
         } else {
-            getOverrideFieldAccessor().setByte(obj, b);
+            setFinal(Reflection.getCallerClass(), obj, () -> fa.setByte(obj, b));
         }
```

Comment:
```suggestion
        if (!Modifier.isFinal(modifiers)) {
            fa.setByte(obj, b);
        } else {
            setFinal(Reflection.getCallerClass(), obj, () -> getOverrideFieldAccessor().setByte(obj, b));
        }
```

Reply:
We still need to capture `b` here so this doesn't really improve anything. Same for others.